server/routes/application-insights.js

```diff
@@ -0,0 +1,38 @@
+const express = require('express');
+const request = require('xhr-request');
+const router = new express.Router();
+
+const host = 'api.applicationinsights.io';
+
+router.post('/query', (req, res) => {
+  const { apiKey, appId, query, queryTimespan } = req.body;
+
+  if (!apiKey || !appId) {
+    return res.send({ error: 'Invalid request parameters' });
```

Comment:
The "if" explicitly checks for apikey and appid, therefore the error should be specific as well.